govcd/openapi.go

```diff
@@ -120,48 +120,58 @@ func (client *Client) OpenApiGetAllItems(apiVersion string, urlRef *url.URL, que
 // returned this function returns "ErrorEntityNotFound: API_ERROR" so that one can use ContainsNotFound(err) to
 // differentiate when an objects was not found from any other error.
 func (client *Client) OpenApiGetItem(apiVersion string, urlRef *url.URL, params url.Values, outType interface{}, additionalHeader map[string]string) error {
+	_, err := client.OpenApiGetItemAndHeader(apiVersion, urlRef, params, outType, additionalHeader, "")
+	return err
+}
+
+// OpenApiGetItemAndHeader is a low level OpenAPI client function to perform GET request for any item and return a given header.
+// The urlRef must point to ID of exact item (e.g. '/1.0.0/edgeGateways/{EDGE_ID}')
+// It responds with HTTP 403: Forbidden - If the user is not authorized or the entity does not exist. When HTTP 403 is
+// returned this function returns "ErrorEntityNotFound: API_ERROR" so that one can use ContainsNotFound(err) to
+// differentiate when an objects was not found from any other error.
+func (client *Client) OpenApiGetItemAndHeader(apiVersion string, urlRef *url.URL, params url.Values, outType interface{}, additionalHeader map[string]string, headerToGet string) (string, error) {
```

Comment:
As the `OpenApiX` are very low-level functions for all OpenAPI matters - should we make it more flexible by making a function that gets all headers in one go (we can later get specific headers for the ones who need it) to avoid people calling it multiple times if such a need arises?

Reply:
That's pretty right, I'm not sure why I didn't do that in the first place, but I've refactored them and now they return all headers for convenience.